Authentication Header Stripped in Requests via Unauthenticated Paths

**Describe the bug**
The authentication header appears to be erroneously stripped or rewritten in requests communicated via paths defined in a provider's unauthenticated paths field.

**To Reproduce**
Steps to reproduce the behavior:

1. Set up a service that requires API access authenticated via the authentication header. For my issue, I set up the docker service [Miniflux](https://hub.docker.com/r/miniflux/miniflux/). This application uses the Google Reader API which employs a custom authentication header auth scheme.
2. Set up a forward auth (single application) proxy provider for the service in Authentik.
3. Add the Google Reader API required paths to the unauthenticated paths field when setting up the authentik provider (for Miniflux, it would be ^/accounts/ClientLogin.* and ^/reader/api/.*
4. Attempt to communicate with the service via the unauthenticated paths. For miniflux, you can do the following steps:
5. Set up a Google Reader API username and password in Miniflux's integrations settings.
6. Execute the following command: curl -div 'https://<your service's URL>/accounts/ClientLogin?Email=<Your Google Reader API username>&Passwd=<Your Google Reader API password>'
7. Get the "Auth" token from the previous command's response and edit the following command with it and then execute it:
curl -iv -H "Authorization: GoogleLogin auth=<your auth token>" 'https://<your service's URL>/reader/api/0/user-info?output=json'
8. Observe the response. It should be an unauthorized response. You can also check the logs in Miniflux and you will see the error "No token provided."


**Expected behavior**
API access should work via unauthenticated paths, even if they send an authentication header. In the case of my example using Miniflux, the server should provide a response to the last command denoting information about the user.

**Screenshots**
<img width="757" alt="image" src="https://github.com/goauthentik/authentik/assets/17125911/b4c38d28-6300-498d-ac91-e85ae08901d0">


**Logs**
Authentik logs show no change when executing the curl commands above. Miniflux shows the following (with personal info sanitized using <description of info> template.)

```

06/03/2023 11:19:52 PM
[2023-06-03T23:19:52] [INFO] [GoogleReader][Login] [ClientIP=<myIP>] User authenticated: <user>
06/03/2023 11:19:52 PM
[2023-06-03T23:19:52] [INFO] [GoogleReader][Login] [ClientIP=<myIP>] Created token: <user>/<token>
06/03/2023 11:20:05 PM
[2023-06-03T23:20:05] [DEBUG] [HTTPS] <myIP> GET /reader/api/0/user-info?output=json
06/03/2023 11:20:05 PM
[2023-06-03T23:20:05] [ERROR] [GoogleReader][Auth] [ClientIP=<myIP>] No token provided
06/03/2023 11:20:05 PM
[2023-06-03T23:20:05] [ERROR] [HTTP:Unauthorized] /reader/api/0/user-info?output=json

```

**Version and Deployment (please complete the following information):**

-   authentik version: 2023.5.3
-   Deployment: Unraid via Docker (not docker-compose)

**Additional context**
If I remove the authentication header from the authResponseHeaders in my reverse proxy (Traefik,) the API exposed on the unauthenticated paths starts working (in this case, Google Reader API.) Adding the header back into the authResponseHeaders causes the issue to return.

This issue was first opened in the [Authentik discord](https://discord.com/channels/809154715984199690/1110986265668616212) and additional information can be found there.

--- a/internal/outpost/proxyv2/application/mode_common.go
+++ b/internal/outpost/proxyv2/application/mode_common.go
@@ -13,6 +13,30 @@ import (
 	"goauthentik.io/internal/constants"
 )
 
+// Attempt to set basic auth based on user's attributes
+func (a *Application) setAuthorizationHeader(headers http.Header, c *Claims) {
+	if !*a.proxyConfig.BasicAuthEnabled {
+		return
+	}
+	userAttributes := c.Proxy.UserAttributes
+	var ok bool
+	var password string
+	if password, ok = userAttributes[*a.proxyConfig.BasicAuthPasswordAttribute].(string); !ok {
+		password = ""
+	}
+	// Check if we should use email or a custom attribute as username
+	var username string
+	if username, ok = userAttributes[*a.proxyConfig.BasicAuthUserAttribute].(string); !ok {
+		username = c.Email
+	}
+	if username == "" && password == "" {
+		return
+	}
+	authVal := base64.StdEncoding.EncodeToString([]byte(username + ":" + password))
+	a.log.WithField("username", username).Trace("setting http basic auth")
+	headers.Set("Authorization", fmt.Sprintf("Basic %s", authVal))
+}
+
 func (a *Application) addHeaders(headers http.Header, c *Claims) {
 	// https://goauthentik.io/docs/providers/proxy/proxy
 	headers.Set("X-authentik-username", c.PreferredUsername)
@@ -33,22 +57,7 @@ func (a *Application) addHeaders(headers http.Header, c *Claims) {
 		return
 	}
 	userAttributes := c.Proxy.UserAttributes
-	// Attempt to set basic auth based on user's attributes
-	if *a.proxyConfig.BasicAuthEnabled {
-		var ok bool
-		var password string
-		if password, ok = userAttributes[*a.proxyConfig.BasicAuthPasswordAttribute].(string); !ok {
-			password = ""
-		}
-		// Check if we should use email or a custom attribute as username
-		var username string
-		if username, ok = userAttributes[*a.proxyConfig.BasicAuthUserAttribute].(string); !ok {
-			username = c.Email
-		}
-		authVal := base64.StdEncoding.EncodeToString([]byte(username + ":" + password))
-		a.log.WithField("username", username).Trace("setting http basic auth")
-		headers.Set("Authorization", fmt.Sprintf("Basic %s", authVal))
-	}
+	a.setAuthorizationHeader(headers, c)
 	// Check if user has additional headers set that we should sent
 	if additionalHeaders, ok := userAttributes["additionalHeaders"].(map[string]interface{}); ok {
 		a.log.WithField("headers", additionalHeaders).Trace("setting additional headers")

